spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkPlanUtil.java

```diff
@@ -53,6 +58,49 @@ private static SparkPlan actualPlan(SparkPlan plan) {
     }
   }
 
+  public static List<Expression> collectExprs(
+      SparkPlan sparkPlan, Predicate<Expression> predicate) {
+    Seq<List<Expression>> seq =
+        SPARK_HELPER.collect(
```

Comment:
Similar to logic in `PlanUtils` but it accounts for AQE plans by relying on `SPARK_HELPER`.
This class also existed before `PlanUtils`, so we might want to converge in the future.

Reply:
If these are only used in tests, should we add them to test utils?

Reply:
This is already a test util, just the one that existed before `PlanUtils`. I would probably merge `PlanUtils` with this class in the future because this one handles AQE plans.